### ISSUE TITLE
No useful output when running bacon pig run --jsonOutput

### ISSUE THREAD

**Issue by goldmann** [project-ncl/bacon#313]
When the `bacon pig run DIR --jsonOutput` call is executed, it should return meaningful data so that it would be possible to identify builds executed in PNC. It is expected to return group build information as well as build information for all executed (and not required builds as well!). In case build should not be required, there should be information provided which build satisfied this request.

This is related to #289, but it talks specifically about returning correct output that can be parsed in this call whereas the #289 is about generally supporting `--jsonOutput` switch.

**Comment by goldmann** [comment]
This is a critical piece that should be part of 2.0.0 release.

**Comment by thescouser89** [comment]
@goldmann Is the output a tad more useful now?

```bash
$ java -jar cli/target/bacon.jar pig run \
      /home/dcheung/projects/work/build-configurations/empty \
      --skipBuilds --skipSharedContent --skipJavadoc --skipSources  \
      --skipRepo --skipBranchCheck

[lots of text omitted]
[INFO ] - PiG run completed, the results are in: /home/dcheung/projects/work/bacon/target
---
empty:
  internalScmUrl: "<gerrit>/michalszynkiewicz/empty.git"
  scmRevision: "52d5b89bbcc39e77f09c4a4ca070f75a41b55819"
  id: "1968"
  name: "empty"
  buildLog: null
  builtArtifacts:
  - gapv: "com.github.michalszynkiewicz.test:empty:jar:version"
    fileName: "empty-version.jar"
    downloadUrl: "<indy instance>api/version.jar"
  - gapv: "com.github.michalszynkiewicz.test:empty:version"
    fileName: "empty-version.pom"
    downloadUrl: "<indy instance>/test.pom"
  dependencyArtifacts: null
```
Would you also like to see the status?

Note that this output comes with a caveat: I think we are erroneously trying to get the latest builds of the build configs defined, rather than getting the result of the builds of the latest group build. I'll have to double-check if the current logic is erroneous or done deliberately.

**Comment by goldmann** [comment]
It would be more useful yes, but I don't think it's sufficient.

What I would like to have is the **builds listed as an array** as well as the **group build metadata**. I would expect an array of build objects that were executed as well as group build object that would describe the overall status.

Not sure if the root key `empty` is useful. Is it possible to build multiple projects? If yes, then we should preserve it.

`builtArtifacts` may or may not be useful here. I need more properties, like checksums to make use of it. But, saying this - this can be queried if I would have build IDs. Other option is to return the serialized artifact object, that would solve it as well.

**Comment by goldmann** [comment]
One additional thing that would be **very handy** is the full version of the release which is basically this:

https://github.com/project-ncl/bacon/blob/0590814da6ed02e471e4b3fc2bcba36579609677/pig/src/main/java/org/jboss/pnc/bacon/pig/impl/PigContext.java#L82

**Comment by thescouser89** [comment]
I was thinking this over the weekend: does PiG 1.x had that functionality or is that new functionality required to better use bacon/PiG in CPaaS?

**Comment by goldmann** [comment]
In CPaaS we had duplicated some functionality that is already provided by PiG. We used 3 ways to interact with PNC. When Bacon became the official tool to interact with PNC we made a decision that other ways will be dropped and we will use only Bacon and the PiG provided functionality to integrate with PNC.

At the same time, in CPaaS we had two build types that were referring to PNC: `pnc-maven` and `pig-maven`. We took this opportunity to unify this as well and we created a single `pig` build type to interact with PNC 2+. Saying this, the `pnc-maven` build was better implemented when talking about handling artifacts (because PiG didn't provide that information in 1.x as well), so I started searching for a way to provide this information via PiG and as a result the two NCL tickets were created plus this one.

Hope this background makes it better understand our state.
